actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -54,13 +54,21 @@ def callable_cache_key?
       def collection_by_cache_keys
         seed = callable_cache_key? ? @options[:cached] : ->(i) { i }
 
+        key_base = ActiveSupport::Cache::KeyMaster.new(@view.combined_fragment_cache_key(nil)).cache_key
         @collection.each_with_object({}) do |item, hash|
-          hash[expanded_cache_key(seed.call(item))] = item
+          key = ActiveSupport::Cache::KeyMaster.new(key_base)
```

Comment:
If I'm reading this right, it seems like you're re-creating an instance of `ActiveSupport::Cache::KeyMaster.new(key_base)` for each object of the loop, when `key_base` is a static value that's declared outside the loop.

Could you also declare `key` outside the loop and then keep the `key << ...` line inside the loop?

Reply:
No because then we would be using the same object for every entry in the hash and then mutating it. If you do that you get AMAZING performance gains, but that's mostly because you end up with only one item in the hash since all of them are using the same key.